### PR TITLE
Sort imports and requires

### LINKED ISSUE #2701: Auto-alphabetize imports
@Daniel15 suggested at https://github.com/badges/shields/pull/2554#pullrequestreview-190125958:

> In general, does this project have rules about alphabetization? At Facebook we alphabetize `require()` statements and JSX props (the latter enforced using [react/jsx-sort-props](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md) lint rule).

If we can automate this, seems easy enough and will help keep things tidy. I do tend to spend a bit of time putting these in order when I move things around.

### PR DESCRIPTION
**Pull request by paulmelnikow**
This will definitely save time, and ensure more uniformity.

It moves the `createServiceTester()` calls to a different place from where I'd like them, though I'm happy to have them checked by the linter.

~~Ref~~ Closes #2701

### CODE REVIEW

**Comment by calebcartwright** [comment]
Apologies if we've already got this documented somewhere, but as far the conventions on imports we want to use/follow: 

First: external packages (alphabetized by package name)
Second: relative paths (alphabetized by file name) 

With no new lines in between the external/3rd party modules and our own code. Is that correct, and will this new rule will enforce that?

**Comment by paulmelnikow** [comment]
All there is now is a pattern I've tried to maintain. It's not documented or even something I comment on.

After this change it's enforced by the linter, along these lines:

This comes from [the `import/order` rule](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md#importorder-enforce-a-convention-in-module-import-order):

```js
// 1. node "builtin" modules
import fs from 'fs';
import path from 'path';
// 2. "external" modules
import _ from 'lodash';
import chalk from 'chalk';
// 3. "internal" modules
// (if you have configured your path or webpack to handle your internal paths differently)
import foo from 'src/foo';
// 4. modules from a "parent" directory
import foo from '../foo';
import qux from '../../foo/qux';
// 5. "sibling" modules from the same or a sibling's directory
import bar from './bar';
import baz from './bar/baz';
// 6. "index" of the current directory
import main from './';
```

That's ESM syntax, but you get the idea.

There is a better module that handles ESM beautifully [eslint-plugin-simple-import-sort](https://github.com/lydell/eslint-plugin-simple-import-sort) and fully autofixable, but it doesn't work on `require`.

**Comment by calebcartwright** [comment]
Oops I forgot the builtins but yeah that's the order I follow (or at least attempt to follow) in my own projects.

👍

File: core/service-test-runner/service-tester.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 'use strict'
 
 const emojic = require('emojic')
+const trace = require('../base-service/trace')
 const frisby = require('./icedfrisby-no-nock')(
+  // eslint-disable-next-line import/order
   require('icedfrisby-nock')(require('icedfrisby'))
```
Comment (calebcartwright) [inline core/service-test-runner/service-tester.js]: Interesting. I figured eslint would've recognized this automatically and not barked

File: frontend/components/category-headings.spec.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import React from 'react'
 import { shallow } from 'enzyme'
 import { expect } from 'chai'
-import { CategoryHeading, CategoryHeadings } from './category-headings'
 import { H3 } from './common'
+import { CategoryHeading, CategoryHeadings } from './category-headings'
```
Comment (calebcartwright) [inline frontend/components/category-headings.spec.js]: I would've expected the previous order (`./cat..` before `./com...`) but guess I should re-read the rule (I assume eslint made all these changes)

File: core/service-test-runner/service-tester.js (hunk 1 above)
Comment (paulmelnikow) [reply core/service-test-runner/service-tester.js]: Yea, the rule probably could be improved. IMO if there are multiple requires on a line it should only consider the outermost one. 🤷‍♂️

File: frontend/components/category-headings.spec.js (hunk 2 above)
Comment (paulmelnikow) [reply frontend/components/category-headings.spec.js]: I think this rule doesn't alphabetize, unfortunately.

Comment (calebcartwright) [reply frontend/components/category-headings.spec.js]: ahhh that makes so much more sense. i was seriously questioning my mental sorting abilities

**Comment by paulmelnikow** [comment]
Me too. I asked at lydell/eslint-plugin-simple-import-sort#6 but we'll just have to wait until we can use `import` or someone comes along to make this for `require`.

**Comment by Daniel15** [comment]
Nice work! This is great.

**Pull request merged**